apps/src/music/analytics/AnalyticsReporter.ts

```diff
@@ -116,36 +122,65 @@ export default class AnalyticsReporter {
     }
   }
 
-  async initialize(): Promise<void> {
+  private async initialize(): Promise<void> {
+    if (this.initialized) {
+      return;
+    }
     const response = await fetch(API_KEY_ENDPOINT);
     const responseJson = await response.json();
 
     if (!responseJson.key) {
       throw new Error('No key for analytics.');
     }
 
-    return init(responseJson.key, undefined, {minIdLength: 1}).promise;
+    init(responseJson.key);
+    this.initialized = true;
+  }
+
+  isSessionInProgress() {
+    return !!this.session;
   }
 
   setUserProperties(userId: number, userType: string, signInState: string) {
-    if (!this.sessionInProgress) {
+    if (!this.session) {
       this.log('No session in progress');
       return;
     }
 
-    const formattedUserId = this.formatUserId(userId);
-    setUserId(formattedUserId);
+    if (userId) {
+      setUserId(this.formatUserId(userId));
+    }
 
-    this.identifyObj.set('userType', userType);
-    this.identifyObj.set('signInState', signInState);
+    const identifyEvent = new Identify();
+    identifyEvent.set('userType', userType);
+    identifyEvent.set('signInState', signInState);
 
-    identify(this.identifyObj);
+    identify(identifyEvent);
 
     this.log(
       `User properties: userId: ${userId}, userType: ${userType}, signInState: ${signInState}`
     );
   }
 
+  setProjectProperty(
+    property: (typeof trackedProjectProperties)[number],
+    value: string | number | undefined
+  ) {
+    if (!this.session) {
+      this.log('No session in progress');
+      return;
+    }
+
+    const identifyEvent = new Identify();
```

Comment:
Just curious - before this update, I think there was one `Identify` instance being used within this class which was initialized in the constructor. Why are you now creating a new `Identify` instance every time you set a project property?

Reply:
Yeah, I realized there were some issues with using a single Identify object where certain properties weren't being captured consistently. Amplitude's docs always seem to show using a single identify object per identify call, so I just went with that pattern here.

Reply:
Oh interesting - thanks!